plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/VacuumTableProcedure.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.procedure;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.Duration;
+import io.trino.spi.connector.TableProcedureMetadata;
+
+import javax.inject.Provider;
+
+import static io.trino.plugin.base.session.PropertyMetadataUtil.durationProperty;
+import static io.trino.plugin.iceberg.procedure.IcebergTableProcedureId.VACUUM;
+
+public class VacuumTableProcedure
+        implements Provider<TableProcedureMetadata>
+{
+    @Override
+    public TableProcedureMetadata get()
+    {
+        return new TableProcedureMetadata(
+                VACUUM.name(),
+                false,
+                ImmutableList.of(
+                        durationProperty(
+                                "retention_threshold",
+                                "Remove only files older than threshold",
```

Comment:
This controls which _snapshots_ are removed; file age doesn't matter, does it?

BTW
the VACUUM should also clean up files in table directory that were not committed to the table (eg failed writes). The file age would matter there.

Reply:
> This controls which snapshots are removed; file age doesn't matter, does it?

right

>BTW
the VACUUM should also clean up files in table directory that were not committed to the table (eg failed writes). The file age would matter there.

that's a good point, I will probably add test for this

Reply:
I tried to write a test for such a case but it looks like that file is not removed, I am looking into that.

Reply:
So `expireSnapshot` doesn't remove `orphan files`. There is spark procedure in Iceberg for that but I don't see anything similar to `expireSnapshot`. 
Probably it will need to be implemented on trino side.

Reply:
> So `expireSnapshot` doesn't remove `orphan files`.

i'd expect that

> There is spark procedure in Iceberg for that  [...]
> Probably it will need to be implemented on trino side.

Maybe we want to call two procedures?  or indeed on our side.

Reply:
right, thought you mean the api call, like the one you had

Reply:
that would be great, but there is no api call like that :/ which is sad